### PR TITLE
[Merged by Bors] - feat: `Fintype.prod_ite_mem`

### PR DESCRIPTION
**Pull request by YaelDillies**
This specializes the existing `Finset.prod_ite_mem`.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 1]
```diff
@@ -2008,6 +2008,10 @@ lemma prod_ite_eq_ite_exists (p : ι → Prop) [DecidablePred p] (h : ∀ i j, p
 
 variable [DecidableEq ι]
 
+@[to_additive]
+lemma prod_ite_mem (s : Finset ι) (f : ι → α) : ∏ i, (if i ∈ s then f i else 1) = ∏ i ∈ s, f i := by
```
Comment (b-mehta) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:2012]:
```suggestion
lemma prod_ite_mem (s : Finset ι) (f : ι → α) : ∏ i, if i ∈ s then f i else 1 = ∏ i ∈ s, f i := by
```
or not?

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Algebra/Module/BigOperators.lean [hunk 2]
```diff
@@ -54,6 +54,6 @@ lemma sum_piFinset_apply (f : κ → α) (s : Finset κ) (i : ι) :
   classical
   rw [Finset.sum_comp]
   simp only [eval_image_piFinset_const, card_filter_piFinset_const s, ite_smul, zero_smul, smul_sum,
-    sum_ite_mem, inter_self]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/BigOperators.lean:57]: What was this referring to before?

Comment (eric-wieser) [reply Mathlib/Algebra/Module/BigOperators.lean:57]: (at a glance, this looks like a duplicate lemma which has a bad name?)

Comment (b-mehta) [reply Mathlib/Algebra/Module/BigOperators.lean:57]: [Finset.sum_ite_mem](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/BigOperators/Group/Finset.html#Finset.sum_ite_mem)

Comment (b-mehta) [reply Mathlib/Algebra/Module/BigOperators.lean:57]: The new one is `Fintype.`, the old one is `Finset.`, and so with both Fintype and Finset open `sum_ite_mem` becomes ambiguous

Comment (eric-wieser) [reply Mathlib/Algebra/Module/BigOperators.lean:57]: Thanks, edited the description to make this clearer.

File: Mathlib/Algebra/BigOperators/Group/Finset.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/BigOperators/Group/Finset.lean:2012]: No, doesn't work. `if then else` has very low precedence

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**